server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorTests.java

```diff
@@ -1377,6 +1382,56 @@ public void testSortingWithNestedAggregations() throws IOException {
         }
     }
 
+    public void testManySegmentsStillSingleton() throws IOException {
+        NumberFieldType nFt = new NumberFieldType("n", NumberFieldMapper.NumberType.LONG);
+        KeywordFieldType strFt = new KeywordFieldType("str", true, true, Collections.emptyMap());
+        AggregationBuilder builder = new TermsAggregationBuilder("n").field("n")
+            .subAggregation(new TermsAggregationBuilder("str").field("str"));
+        try (Directory directory = newDirectory()) {
+            RandomIndexWriter iw = new RandomIndexWriter(
+                random(),
+                directory,
+                // Attempt to disable merging.
+                LuceneTestCase.newIndexWriterConfig(random(), new StandardAnalyzer()).setMergePolicy(NoMergePolicy.INSTANCE)
```

Comment:
This is handy.  I wouldn't mind a method in `AggregatorTestCase` to get a non-merging(ish) index writer.